File: Easy/Array/2553. Separate the Digits in an Array/Solution.java
class Solution {
    public int[] separateDigits(int[] nums) {
        
        List<Integer> list = new ArrayList<>();

        for(int num : nums) {
            
            String s = Integer.toString(num);

            for(int i = 0; i < s.length(); i++){
                list.add(s.charAt(i) - '0');
            }
        }

        int[] answer = new int[list.size()];
        for(int i = 0; i < list.size(); i++){
            answer[i] = list.get(i);
        }

        return answer;
    }
}

File: Easy/Array/2553. Separate the Digits in an Array/Solution.kt
class Solution {
    fun separateDigits(nums: IntArray): IntArray {
        
        val list = mutableListOf<Int>()

        for(num in nums){

            val digits = num.toString().map {
                it.digitToInt()
            }

            list.addAll(digits)
        }

        return list.toIntArray()
    }
}
